src/components/AssistantTextProcessingForm.vue

```
@@ -315,23 +315,53 @@ export default {
 		console.debug('[assistant] form\'s myoutputs', this.myOutputs)
 	},
 	methods: {
+		// Parse the file if a fileId is passed as initial value to a text field
+		parseTextFileInputs(taskType) {
+			if (taskType === undefined || taskType === null) {
+				return
+			}
+			Object.keys(this.myInputs).forEach(k => {
+				if (taskType.inputShape[k]?.type === 'Text') {
+					if (this.myInputs[k]?.fileId || this.myInputs[k]?.filePath) {
+						const { filePath, fileId } = { fileId: this.myInputs[k]?.fileId, filePath: this.myInputs[k]?.filePath }
+						this.myInputs[k] = ''
+						this.parseFile({ fileId, filePath })
+							.then(response => {
+								if (response.data?.ocs?.data?.parsedText) {
+									this.myInputs[k] = response.data?.ocs?.data?.parsedText
+								}
+							})
```

an empty catch block would be good here

---

Could even send a toast that it failed

---

Done